### ISSUE TITLE
batchNormalization is not implemented for array of rank 5 yet

### ISSUE THREAD

**Issue by prouast** [tensorflow/tfjs#2504]
#### TensorFlow.js version
1.3.1
#### Browser version
Version 78.0.3904.108 (Official Build) (64-bit)
#### Describe the problem or feature request
As the title says, 5D tensors are not supported yet for BatchNorm layers.

This is currently what is stopping me from running a small 3D CNN on video data in tfjs.
I used tfjs.converters.save_keras_model() to save the model which includes a tf.keras.layers.BatchNormalization layer operating on a 5D tensor (see [normalization.ts](https://github.com/tensorflow/tfjs/blob/bec2196c8861470f929a3afee74eef1f10a2258c/tfjs-layers/src/layers/normalization.ts#L66)):

`Error: batchNormalization is not implemented for array of rank 5 yet`

Support for 5D tensors/3D ops has been added to several components recently (#767 #1035 #1985), so I am filing this hoping that someone with free time will consider adding 5D tensor support for BatchNorm too.

**Comment by caisq** [comment]
@dsmilkov @nsthorat FYI, in order to implement rank-5 support for the batchNormalization layer in tfjs-layers, we'll need tf.batchNorm5d() in tfjs-core first. But that doesn't exist yet. 

Adding 'core' tag to this issue.

**Comment by dibenedetto** [comment]
> Closing as stale. Please @mention us if this needs more attention.

@google-ml-butler pinging on the topic.

**Comment by dibenedetto** [comment]
@prouast are there any news?
thank you.

**Comment by prouast** [comment]
This is a two year old feature request which hasn't been getting any attention

**Comment by dibenedetto** [comment]
> @dibenedetto This is a two year old feature request which hasn't been getting any attention.

@prouast it's very strange, i mean, batch normalization for 5d tensor seems quite obvious for anyone who wants to process 3D tensors (e.g., medical 3D images).
did you get any alternative?

**Comment by prouast** [comment]
> > @dibenedetto This is a two year old feature request which hasn't been getting any attention.
> 
> @prouast it's very strange, i mean, batch normalization for 5d tensor seems quite obvious for anyone who wants to process 3D tensors (e.g., medical 3D images). did you get any alternative?

Agreed, which is why I opened this in the first place. I just ended up making an alternative model without batch norm back then.

**Comment by dibenedetto** [comment]
@prouast I'll try to re-open a similar issue.

**Comment by prouast** [comment]
> @prouast I'll try to re-open a similar issue.

Thanks, hopefully it will get more attention this way.
